### DISCUSSION TITLE
Password expiration testing

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8043] [Q&A] [answered]
Hi there!

I am writing an integration test of my authentication solution based on Zitadel. How can I trigger user password expiration without waiting for an actual amount of time to test it?

Thank you.

**Comment by hifabienne** [comment]
hei @ashep You mean that the user has to set a new password after certain time?
If thats the case, ZITADEL doesn't currently provide a possibality to tell that a password should expire after a certain time.

**Reply by ashep** [reply]
Hi @hifabienne! 

I mean that Zitadel has the [max password age policy](https://zitadel.com/docs/apis/resources/mgmt/management-service-get-password-age-policy). If I understand correctly, after `maxAgeDays` value is reached for a user since its last password change, the user's `passwordChangeRequired` [flag](https://zitadel.com/docs/apis/resources/user_service/user-service-get-user-by-id) is automatically set to `true`, right?

The question is: how to artificially trigger the event of password "expiration" and get `passwordChangeRequired` set to `true` because of that?

**Reply by hifabienne** [reply, marked as answer]
As stated in the API docs, the password age settings are not implemented: https://zitadel.com/docs/apis/resources/mgmt/management-service-update-custom-password-age-policy#update-password-age-settings

**Reply by ashep** [reply]
I got it, thank you. I overlooked that. It's so unexpected to see an unimplemented API method in public documentation 😞

**Comment by ashep** [comment]
And another one related question, please. What API can I use to retrieve the last user's password change time?
